armi/nuclearDataIO/geodst.py

```diff
@@ -0,0 +1,298 @@
+                self._geom.bucklings, "float", self._metadata["NBS"]
+            )
+            self._geom.boundaryConstants = record.rwList(
+                self._geom.boundaryConstants, "float", self._metadata["NBCS"]
+            )
+            self._geom.internalBlackBoundaryConstants = record.rwList(
+                self._geom.internalBlackBoundaryConstants,
+                "float",
+                self._metadata["NIBCS"],
+            )
+            self._geom.zonesWithBlackAbs = record.rwList(
+                self._geom.zonesWithBlackAbs, "int", self._metadata["NZWBB"]
+            )
+            self._geom.zoneClassifications = record.rwList(
+                self._geom.zoneClassifications, "int", self._metadata["NZONE"]
+            )
+            self._geom.regionZoneNumber = record.rwList(
+                self._geom.regionZoneNumber, "int", self._metadata["NREG"]
+            )
+
+    def _rw6DRecord(self):
+        """Read/write region assignments to coarse mesh interval."""
+        if self._geom.coarseMeshRegions is None:
+            # initialize all-zeros here before reading now that we
+            # have the matrix dimension metadata available.
+            self._geom.coarseMeshRegions = numpy.zeros(
+                (
+                    self._metadata["NCINTI"],
+                    self._metadata["NCINTJ"],
+                    self._metadata["NCINTK"],
+                ),
+                dtype=numpy.int8,
+            )
+        for ki in range(self._metadata["NCINTK"]):
+            with self.createRecord() as record:
+                # pylint: disable=protected-access
+                self._geom.coarseMeshRegions[:, :, ki] = record._rwMatrix(
+                    self._geom.coarseMeshRegions[:, :, ki],
+                    record.rwInt,
+                    self._metadata["NCINTI"],
+                    self._metadata["NCINTJ"],
+                )
+
+    def _rw7DRecord(self):
+        """Read/write region assignments to fine mesh interval."""
+        if self._geom.fineMeshRegions is None:
+            # initialize all-zeros here before reading now that we
+            # have the matrix dimension metadata available.
+            self._geom.fineMeshRegions = numpy.zeros(
+                (
+                    self._metadata["NINTI"],
+                    self._metadata["NINTJ"],
+                    self._metadata["NINTK"],
+                ),
+                dtype=numpy.int8,
+            )
+        for ki in range(self._metadata["NINTK"]):
+            with self.createRecord() as record:
+                # pylint: disable=protected-access
+                self._geom.fineMeshRegions[:, :, ki] = record._rwMatrix(
+                    self._geom.fineMeshRegions[:, :, ki],
+                    record.rwInt,
+                    self._metadata["NINTI"],
+                    self._metadata["NINTJ"],
+                )
+
+
+readBinary = GeodstStream.readBinary  # pylint: disable=invalid-name
```

Comment:
What's pylint's complaint here? Does it not know to look at the base class?

Reply:
actually it has no problem!